Gateway/Request/CheckoutDataBuilder.php

```diff
@@ -262,7 +282,9 @@ protected function getOpenInvoiceData($order): array
                 'description' => $item->getName(),
                 'quantity' => $numberOfItems,
                 'taxCategory' => $item->getProduct()->getAttributeText('tax_class_id'),
-                'taxPercentage' => $formattedTaxPercentage
+                'taxPercentage' => $formattedTaxPercentage,
+                'productUrl'=> $item->getProduct()->getUrlModel()->getUrl($item->getProduct()),
```

Comment:
Have you tested this with products in different storeviews so that the correct URL is used?